pkg/binary/envoy/debug/pii.go

```diff
@@ -0,0 +1,72 @@
+package debug
+
+import (
+	"fmt"
+	"bitbucket.org/creachadair/shell"
+	"crypto/sha256"
+)
+
+// filter the array of logs using formatStr and validFieldNames, hash the resulting array of logs and return the final array
+func process(logsArr []string, formatStr string, validFieldNames map[string]bool) (processedLogs []string) {
+	defer func() {
+		if r := recover(); r != nil {
+			fmt.Println("process func panicking, recovering")
+		}
+	} ()
+
+	fieldNames, ok := shell.Split(formatStr)
+
+	if !ok {
+		panic(fmt.Sprint("error in splitting format string: %s", formatStr))
+	}
+
+	testForValidLog :=  func(log string) bool {
+		fields, ok := shell.Split(log)
+		if !ok {
+			panic(fmt.Sprint("error in splitting log: %s", log))
+		}
+		// TODO: consider using regex to check fields instead of merely verifiying the number of fields
+		return len(fields) == len(fieldNames)
+	}
+	validLogs := filter(logsArr, testForValidLog)
+
```

Comment:
Here is where we should declare our output variable. Traditionally, we'd either call this `out`, or something that made sense semantically. Your `processedLogs` is on the right track, but too long: it should be either `processed` or `out`.

```go
processed := make([]string, 0, len(validLogs))
```

Note that declaring it here means we can correctly-size the array from the get-go: we know exactly how many elements will be in it, so we can allocate enough memory up front and avoid having to copy the array while we're in this loop.